utils/swift_build_support/swift_build_support/products/product.py

```diff
@@ -191,24 +191,19 @@ def is_release(self):
         """
         return is_release_variant(self.args.build_variant)
 
-    def install_toolchain_path(self, host_target, force_local=False):
+    def install_toolchain_path(self, host_target):
         """toolchain_path() -> string
 
-        Returns the path to the local toolchain that is being created as part of
-        this build, or to a native prebuilt toolchain that was passed in.
+        Returns the path to the toolchain that is being created as part of this
+        build
         """
-        if self.args.native_swift_tools_path is not None and not force_local:
-            return os.path.split(self.args.native_swift_tools_path)[0]
-
         install_destdir = self.args.install_destdir
         if self.args.cross_compile_hosts:
             if targets.StdlibDeploymentTarget.get_target_for_name(
                     host_target).platform.is_darwin:
                 build_root = os.path.dirname(self.build_dir)
                 install_destdir = '%s/intermediate-install/%s' % (build_root,
                                                                   host_target)
-            elif host_target != self.args.host_target:
-                install_destdir = os.path.join(install_destdir, host_target)
```

Comment:
This removed line is the path to a cross-compiled toolchain, but this function is used by build products expecting a native compiler. It worked fine for me because I never hit this, since I always used the prebuilt toolchain path above, but this line would've broken scenario 2.